==> diffs/export_invalid_mutli.diff <==
diff of execution of:
echo 'export A------=a 123=4 B=b MINUS-' | tr ';' '\n' | [shell]
--- expected	2024-09-01 20:04:09.780678560 +0200
+++ found	2024-09-01 20:04:09.856678633 +0200
@@ -1,7 +1,5 @@
 
 ======== stderr =======
 [progname]: export: `A------=a': not a valid identifier
-[progname]: export: `123=4': not a valid identifier
-[progname]: export: `MINUS-': not a valid identifier
 ======== return =======
 return value: 1

==> diffs/here_doc.diff <==
diff of execution of:
echo 'cat << EOF;hi;ha;ho;EOF' | tr ';' '\n' | [shell]
--- expected	2024-09-01 20:04:17.954686334 +0200
+++ found	2024-09-01 20:04:18.029686405 +0200
@@ -1,8 +1,11 @@
 hi
 ha
 ho
 
 ======== stderr =======
-
+[progname]: hi: command not found
+[progname]: ha: command not found
+[progname]: ho: command not found
+[progname]: EOF: command not found
 ======== return =======
 return value: 0

==> diffs/here_doc_multivar.diff <==
diff of execution of:
echo 'cat <<EOF; $PWD and $OLDPWD;ha;$PATH ;EOF' | tr ';' '\n' | [shell]
--- expected	2024-09-01 20:04:18.549686894 +0200
+++ found	2024-09-01 20:04:18.628686969 +0200
@@ -1,8 +1,11 @@
- /home/pcasagra/42/minishell_brunos_branch/test/temp and /home/pcasagra/42/minishell_brunos_branch/test
+ $PWD and $OLDPWD
 ha
-/usr/local/nodejs/bin:/home/pcasagra/.local/bin:/usr/local/nodejs/bin:/home/pcasagra/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin:/home/pcasagra/clion-2024.1.2/bin/clion.sh:/home/pcasagra/clion-2024.1.2/bin/clion.sh 
+$PATH 
 
 ======== stderr =======
-
+[progname]: /home/pcasagra/42/minishell_brunos_branch/test/temp: Is a directory
+[progname]: ha: command not found
+[progname]: /usr/local/nodejs/bin:/home/pcasagra/.local/bin:/usr/local/nodejs/bin:/home/pcasagra/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin:/home/pcasagra/clion-2024.1.2/bin/clion.sh:/home/pcasagra/clion-2024.1.2/bin/clion.sh: No such file or directory
+[progname]: EOF: command not found
 ======== return =======
 return value: 0

==> diffs/here_doc_quotes.diff <==
diff of execution of:
echo 'cat << EOF;\'$PWD\';EOF' | tr ';' '\n' | [shell]
--- expected	2024-09-01 20:04:18.328686686 +0200
+++ found	2024-09-01 20:04:18.420686773 +0200
@@ -1,6 +1,7 @@
-'/home/pcasagra/42/minishell_brunos_branch/test/temp'
+'$PWD'
 
 ======== stderr =======
-
+[progname]: $PWD: command not found
+[progname]: EOF: command not found
 ======== return =======
 return value: 0

==> diffs/here_doc_status1.diff <==
diff of execution of:
echo 'cat << EOF;$?;EOF' | tr ';' '\n' | [shell]
--- expected	2024-09-01 20:04:18.834687163 +0200
+++ found	2024-09-01 20:04:18.964687285 +0200
@@ -1,6 +1,7 @@
-0
+$?
 
 ======== stderr =======
-
+[progname]: 0: command not found
+[progname]: EOF: command not found
 ======== return =======
 return value: 0

==> diffs/here_doc_status2.diff <==
diff of execution of:
echo 'cat fail; cat << EOF;$?;EOF' | tr ';' '\n' | [shell]
--- expected	2024-09-01 20:04:19.036687353 +0200
+++ found	2024-09-01 20:04:19.105687418 +0200
@@ -1,6 +1,8 @@
-1
+$?
 
 ======== stderr =======
 [progname]: fail: No such file or directory
+[progname]: 0: command not found
+[progname]: EOF: command not found
 ======== return =======
 return value: 0

==> diffs/here_doc_var.diff <==
diff of execution of:
echo 'cat << EOF;$PWD;EOF' | tr ';' '\n' | [shell]
--- expected	2024-09-01 20:04:18.114686485 +0200
+++ found	2024-09-01 20:04:18.216686581 +0200
@@ -1,6 +1,7 @@
-/home/pcasagra/42/minishell_brunos_branch/test/temp
+$PWD
 
 ======== stderr =======
-
+[progname]: /home/pcasagra/42/minishell_brunos_branch/test/temp: Is a directory
+[progname]: EOF: command not found
 ======== return =======
 return value: 0

==> diffs/path_add_local.diff <==
diff of execution of:
echo 'export PATH=$PATH\':./\'; echo $PATH; infile' | tr ';' '\n' | [shell]
--- expected	2024-09-01 20:04:02.251671287 +0200
+++ found	2024-09-01 20:04:02.293671328 +0200
@@ -1,6 +1,6 @@
-/usr/local/nodejs/bin:/home/pcasagra/.local/bin:/usr/local/nodejs/bin:/home/pcasagra/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin:/home/pcasagra/clion-2024.1.2/bin/clion.sh:/home/pcasagra/clion-2024.1.2/bin/clion.sh:./
+
 
 ======== stderr =======
 [progname]: infile: Permission denied
 ======== return =======
 return value: 0

==> diffs/path_empty_part.diff <==
diff of execution of:
echo 'export PATH=$PATH\':::::/hehe\'; echo $PATH; cat infile' | tr ';' '\n' | [shell]
--- expected	2024-09-01 20:04:03.377672381 +0200
+++ found	2024-09-01 20:04:03.428672431 +0200
@@ -1,8 +1,6 @@
-/usr/local/nodejs/bin:/home/pcasagra/.local/bin:/usr/local/nodejs/bin:/home/pcasagra/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin:/home/pcasagra/clion-2024.1.2/bin/clion.sh:/home/pcasagra/clion-2024.1.2/bin/clion.sh:::::/hehe
-infile content
-infile second line
 
-======== stderr =======
 
+======== stderr =======
+[progname]: cat: No such file or directory
 ======== return =======
 return value: 0

==> diffs/path_empty_part_local.diff <==
diff of execution of:
echo 'export PATH=$PATH\':::::/hehe\'; echo $PATH; infile' | tr ';' '\n' | [shell]
--- expected	2024-09-01 20:04:03.526672526 +0200
+++ found	2024-09-01 20:04:03.597672595 +0200
@@ -1,6 +1,6 @@
-/usr/local/nodejs/bin:/home/pcasagra/.local/bin:/usr/local/nodejs/bin:/home/pcasagra/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin:/home/pcasagra/clion-2024.1.2/bin/clion.sh:/home/pcasagra/clion-2024.1.2/bin/clion.sh:::::/hehe
+
 
 ======== stderr =======
 [progname]: infile: Permission denied
 ======== return =======
 return value: 0

==> diffs/questionable_exec_dir.diff <==
diff of execution of:
echo 'export PATH="$PATH:$PWD"; mkdir asdfg; asdfg' | tr ';' '\n' | [shell]
--- expected	2024-09-01 20:04:00.429669510 +0200
+++ found	2024-09-01 20:04:00.487669566 +0200
@@ -1,5 +1,6 @@
 
 ======== stderr =======
+[progname]: mkdir: command not found
 [progname]: asdfg: command not found
 ======== return =======
 return value: 1

==> diffs/questionable_exec_dot.diff <==
diff of execution of:
echo '.' | tr ';' '\n' | [shell]
--- expected	2024-09-01 20:03:59.919669012 +0200
+++ found	2024-09-01 20:03:59.960669052 +0200
@@ -1,6 +1,5 @@
 
 ======== stderr =======
-[progname]: .: filename argument required
-[progname]: usage: . filename [arguments]
+[progname]: .: Is a directory
 ======== return =======
 return value: 1

==> diffs/questionable_exec_dotdot.diff <==
diff of execution of:
echo '..' | tr ';' '\n' | [shell]
--- expected	2024-09-01 20:04:00.073669162 +0200
+++ found	2024-09-01 20:04:00.155669242 +0200
@@ -1,5 +1,5 @@
 
 ======== stderr =======
-[progname]: ..: command not found
+[progname]: ..: Is a directory
 ======== return =======
 return value: 1

==> diffs/questionable_export.diff <==
diff of execution of:
echo 'export' | tr ';' '\n' | [shell]
--- expected	2024-09-01 20:03:58.615667735 +0200
+++ found	2024-09-01 20:03:58.707667825 +0200
@@ -1,72 +1,72 @@
-declare -x CHROME_DESKTOP="code-url-handler.desktop"
-declare -x COLORTERM="truecolor"
-declare -x DBUS_SESSION_BUS_ADDRESS="unix:path=/run/user/1000/bus"
-declare -x DESKTOP_SESSION="ubuntu"
-declare -x DISPLAY=":0"
-declare -x GDK_BACKEND="x11"
-declare -x GDMSESSION="ubuntu"
-declare -x GIO_LAUNCHED_DESKTOP_FILE="/usr/share/applications/code.desktop"
-declare -x GIO_LAUNCHED_DESKTOP_FILE_PID="2342"
-declare -x GIT_ASKPASS="/usr/share/code/resources/app/extensions/git/dist/askpass.sh"
-declare -x GJS_DEBUG_OUTPUT="stderr"
-declare -x GJS_DEBUG_TOPICS="JS ERROR;JS LOG"
-declare -x GNOME_DESKTOP_SESSION_ID="this-is-deprecated"
-declare -x GNOME_SETUP_DISPLAY=":1"
-declare -x GNOME_SHELL_SESSION_MODE="ubuntu"
-declare -x GTK_MODULES="gail:atk-bridge"
-declare -x HOME="/home/pcasagra"
-declare -x IM_CONFIG_CHECK_ENV="1"
-declare -x IM_CONFIG_PHASE="1"
-declare -x INVOCATION_ID="e94f03fe0a6c4e15909f802cd253c84f"
-declare -x JOURNAL_STREAM="8:7897"
-declare -x LANG="en_US.UTF-8"
-declare -x LC_ADDRESS="de_DE.UTF-8"
-declare -x LC_IDENTIFICATION="de_DE.UTF-8"
-declare -x LC_MEASUREMENT="de_DE.UTF-8"
-declare -x LC_MONETARY="de_DE.UTF-8"
-declare -x LC_NAME="de_DE.UTF-8"
-declare -x LC_NUMERIC="de_DE.UTF-8"
-declare -x LC_PAPER="de_DE.UTF-8"
-declare -x LC_TELEPHONE="de_DE.UTF-8"
-declare -x LC_TIME="de_DE.UTF-8"
-declare -x LESSCLOSE="/usr/bin/lesspipe %s %s"
-declare -x LESSOPEN="| /usr/bin/lesspipe %s"
-declare -x LOGNAME="pcasagra"
-declare -x LS_COLORS="rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.zst=01;31:*.tzst=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.wim=01;31:*.swm=01;31:*.dwm=01;31:*.esd=01;31:*.jpg=01;35:*.jpeg=01;35:*.mjpg=01;35:*.mjpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.webp=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36:"
-declare -x MANAGERPID="1201"
-declare -x OLDPWD="/home/pcasagra/42/minishell_brunos_branch/test"
-declare -x ORIGINAL_XDG_CURRENT_DESKTOP="ubuntu:GNOME"
-declare -x PATH="/usr/local/nodejs/bin:/home/pcasagra/.local/bin:/usr/local/nodejs/bin:/home/pcasagra/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin:/home/pcasagra/clion-2024.1.2/bin/clion.sh:/home/pcasagra/clion-2024.1.2/bin/clion.sh"
-declare -x PWD="/home/pcasagra/42/minishell_brunos_branch/test/temp"
-declare -x QT_ACCESSIBILITY="1"
-declare -x QT_IM_MODULE="ibus"
-declare -x SESSION_MANAGER="local/pcasagra-VirtualBox:@/tmp/.ICE-unix/1346,unix/pcasagra-VirtualBox:/tmp/.ICE-unix/1346"
-declare -x SHELL="/bin/bash"
-declare -x SSH_AGENT_LAUNCHER="gnome-keyring"
-declare -x SSH_AUTH_SOCK="/run/user/1000/keyring/ssh"
-declare -x SYSTEMD_EXEC_PID="1383"
-declare -x TERM="xterm-256color"
-declare -x TERM_PROGRAM="vscode"
-declare -x TERM_PROGRAM_VERSION="1.90.0"
-declare -x USER="pcasagra"
-declare -x USERNAME="pcasagra"
-declare -x VSCODE_GIT_ASKPASS_EXTRA_ARGS=""
-declare -x VSCODE_GIT_ASKPASS_MAIN="/usr/share/code/resources/app/extensions/git/dist/askpass-main.js"
-declare -x VSCODE_GIT_ASKPASS_NODE="/usr/share/code/code"
-declare -x VSCODE_GIT_IPC_HANDLE="/run/user/1000/vscode-git-806976d2ff.sock"
-declare -x WAYLAND_DISPLAY="wayland-0"
-declare -x XAUTHORITY="/run/user/1000/.mutter-Xwaylandauth.RHKKT2"
-declare -x XDG_CONFIG_DIRS="/etc/xdg/xdg-ubuntu:/etc/xdg"
-declare -x XDG_CURRENT_DESKTOP="Unity"
-declare -x XDG_DATA_DIRS="/usr/share/ubuntu:/usr/local/share/:/usr/share/:/var/lib/snapd/desktop"
-declare -x XDG_MENU_PREFIX="gnome-"
-declare -x XDG_RUNTIME_DIR="/run/user/1000"
-declare -x XDG_SESSION_CLASS="user"
-declare -x XDG_SESSION_DESKTOP="ubuntu"
-declare -x XDG_SESSION_TYPE="wayland"
-declare -x XMODIFIERS="@im=ibus"
+declare -x SHELL=/bin/bash
+declare -x SESSION_MANAGER=local/pcasagra-VirtualBox:@/tmp/.ICE-unix/1346,unix/pcasagra-VirtualBox:/tmp/.ICE-unix/1346
+declare -x QT_ACCESSIBILITY=1
+declare -x COLORTERM=truecolor
+declare -x XDG_CONFIG_DIRS=/etc/xdg/xdg-ubuntu:/etc/xdg
+declare -x SSH_AGENT_LAUNCHER=gnome-keyring
+declare -x XDG_MENU_PREFIX=gnome-
+declare -x TERM_PROGRAM_VERSION=1.90.0
+declare -x GNOME_DESKTOP_SESSION_ID=this-is-deprecated
+declare -x LC_ADDRESS=de_DE.UTF-8
+declare -x GNOME_SHELL_SESSION_MODE=ubuntu
+declare -x LC_NAME=de_DE.UTF-8
+declare -x SSH_AUTH_SOCK=/run/user/1000/keyring/ssh
+declare -x XMODIFIERS=@im=ibus
+declare -x DESKTOP_SESSION=ubuntu
+declare -x LC_MONETARY=de_DE.UTF-8
+declare -x GTK_MODULES=gail:atk-bridge
+declare -x PWD=/home/pcasagra/42/minishell_brunos_branch/test/temp
+declare -x LOGNAME=pcasagra
+declare -x XDG_SESSION_DESKTOP=ubuntu
+declare -x XDG_SESSION_TYPE=wayland
+declare -x SYSTEMD_EXEC_PID=1383
+declare -x XAUTHORITY=/run/user/1000/.mutter-Xwaylandauth.RHKKT2
+declare -x VSCODE_GIT_ASKPASS_NODE=/usr/share/code/code
+declare -x IM_CONFIG_CHECK_ENV=1
+declare -x GJS_DEBUG_TOPICS=JS ERROR;JS LOG
+declare -x HOME=/home/pcasagra
+declare -x USERNAME=pcasagra
+declare -x IM_CONFIG_PHASE=1
+declare -x LANG=en_US.UTF-8
+declare -x LC_PAPER=de_DE.UTF-8
+declare -x LS_COLORS=rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.zst=01;31:*.tzst=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.wim=01;31:*.swm=01;31:*.dwm=01;31:*.esd=01;31:*.jpg=01;35:*.jpeg=01;35:*.mjpg=01;35:*.mjpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.webp=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36:
+declare -x XDG_CURRENT_DESKTOP=Unity
+declare -x WAYLAND_DISPLAY=wayland-0
+declare -x GIT_ASKPASS=/usr/share/code/resources/app/extensions/git/dist/askpass.sh
+declare -x INVOCATION_ID=e94f03fe0a6c4e15909f802cd253c84f
+declare -x MANAGERPID=1201
+declare -x CHROME_DESKTOP=code-url-handler.desktop
+declare -x GJS_DEBUG_OUTPUT=stderr
+declare -x VSCODE_GIT_ASKPASS_EXTRA_ARGS=
+declare -x GNOME_SETUP_DISPLAY=:1
+declare -x LESSCLOSE=/usr/bin/lesspipe %s %s
+declare -x XDG_SESSION_CLASS=user
+declare -x LC_IDENTIFICATION=de_DE.UTF-8
+declare -x TERM=xterm-256color
+declare -x LESSOPEN=| /usr/bin/lesspipe %s
+declare -x USER=pcasagra
+declare -x VSCODE_GIT_IPC_HANDLE=/run/user/1000/vscode-git-806976d2ff.sock
+declare -x DISPLAY=:0
+declare -x LC_TELEPHONE=de_DE.UTF-8
+declare -x QT_IM_MODULE=ibus
+declare -x LC_MEASUREMENT=de_DE.UTF-8
+declare -x XDG_RUNTIME_DIR=/run/user/1000
+declare -x LC_TIME=de_DE.UTF-8
+declare -x VSCODE_GIT_ASKPASS_MAIN=/usr/share/code/resources/app/extensions/git/dist/askpass-main.js
+declare -x JOURNAL_STREAM=8:7897
+declare -x XDG_DATA_DIRS=/usr/share/ubuntu:/usr/local/share/:/usr/share/:/var/lib/snapd/desktop
+declare -x GDK_BACKEND=x11
+declare -x PATH=/usr/local/nodejs/bin:/home/pcasagra/.local/bin:/usr/local/nodejs/bin:/home/pcasagra/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin:/home/pcasagra/clion-2024.1.2/bin/clion.sh:/home/pcasagra/clion-2024.1.2/bin/clion.sh
+declare -x GDMSESSION=ubuntu
+declare -x ORIGINAL_XDG_CURRENT_DESKTOP=ubuntu:GNOME
+declare -x DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/1000/bus
+declare -x GIO_LAUNCHED_DESKTOP_FILE_PID=2342
+declare -x GIO_LAUNCHED_DESKTOP_FILE=/usr/share/applications/code.desktop
+declare -x LC_NUMERIC=de_DE.UTF-8
+declare -x OLDPWD=/home/pcasagra/42/minishell_brunos_branch/test
+declare -x TERM_PROGRAM=vscode
 
 ======== stderr =======
 
 ======== return =======
 return value: 0

==> diffs/questionable_export_quotes.diff <==
diff of execution of:
echo 'export a=\'"quotes"\'; export' | tr ';' '\n' | [shell]
--- expected	2024-09-01 20:03:58.798667915 +0200
+++ found	2024-09-01 20:03:58.840667956 +0200
@@ -1,73 +1,73 @@
-declare -x CHROME_DESKTOP="code-url-handler.desktop"
-declare -x COLORTERM="truecolor"
-declare -x DBUS_SESSION_BUS_ADDRESS="unix:path=/run/user/1000/bus"
-declare -x DESKTOP_SESSION="ubuntu"
-declare -x DISPLAY=":0"
-declare -x GDK_BACKEND="x11"
-declare -x GDMSESSION="ubuntu"
-declare -x GIO_LAUNCHED_DESKTOP_FILE="/usr/share/applications/code.desktop"
-declare -x GIO_LAUNCHED_DESKTOP_FILE_PID="2342"
-declare -x GIT_ASKPASS="/usr/share/code/resources/app/extensions/git/dist/askpass.sh"
-declare -x GJS_DEBUG_OUTPUT="stderr"
-declare -x GJS_DEBUG_TOPICS="JS ERROR;JS LOG"
-declare -x GNOME_DESKTOP_SESSION_ID="this-is-deprecated"
-declare -x GNOME_SETUP_DISPLAY=":1"
-declare -x GNOME_SHELL_SESSION_MODE="ubuntu"
-declare -x GTK_MODULES="gail:atk-bridge"
-declare -x HOME="/home/pcasagra"
-declare -x IM_CONFIG_CHECK_ENV="1"
-declare -x IM_CONFIG_PHASE="1"
-declare -x INVOCATION_ID="e94f03fe0a6c4e15909f802cd253c84f"
-declare -x JOURNAL_STREAM="8:7897"
-declare -x LANG="en_US.UTF-8"
-declare -x LC_ADDRESS="de_DE.UTF-8"
-declare -x LC_IDENTIFICATION="de_DE.UTF-8"
-declare -x LC_MEASUREMENT="de_DE.UTF-8"
-declare -x LC_MONETARY="de_DE.UTF-8"
-declare -x LC_NAME="de_DE.UTF-8"
-declare -x LC_NUMERIC="de_DE.UTF-8"
-declare -x LC_PAPER="de_DE.UTF-8"
-declare -x LC_TELEPHONE="de_DE.UTF-8"
-declare -x LC_TIME="de_DE.UTF-8"
-declare -x LESSCLOSE="/usr/bin/lesspipe %s %s"
-declare -x LESSOPEN="| /usr/bin/lesspipe %s"
-declare -x LOGNAME="pcasagra"
-declare -x LS_COLORS="rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.zst=01;31:*.tzst=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.wim=01;31:*.swm=01;31:*.dwm=01;31:*.esd=01;31:*.jpg=01;35:*.jpeg=01;35:*.mjpg=01;35:*.mjpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.webp=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36:"
-declare -x MANAGERPID="1201"
-declare -x OLDPWD="/home/pcasagra/42/minishell_brunos_branch/test"
-declare -x ORIGINAL_XDG_CURRENT_DESKTOP="ubuntu:GNOME"
-declare -x PATH="/usr/local/nodejs/bin:/home/pcasagra/.local/bin:/usr/local/nodejs/bin:/home/pcasagra/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin:/home/pcasagra/clion-2024.1.2/bin/clion.sh:/home/pcasagra/clion-2024.1.2/bin/clion.sh"
-declare -x PWD="/home/pcasagra/42/minishell_brunos_branch/test/temp"
-declare -x QT_ACCESSIBILITY="1"
-declare -x QT_IM_MODULE="ibus"
-declare -x SESSION_MANAGER="local/pcasagra-VirtualBox:@/tmp/.ICE-unix/1346,unix/pcasagra-VirtualBox:/tmp/.ICE-unix/1346"
-declare -x SHELL="/bin/bash"
-declare -x SSH_AGENT_LAUNCHER="gnome-keyring"
-declare -x SSH_AUTH_SOCK="/run/user/1000/keyring/ssh"
-declare -x SYSTEMD_EXEC_PID="1383"
-declare -x TERM="xterm-256color"
-declare -x TERM_PROGRAM="vscode"
-declare -x TERM_PROGRAM_VERSION="1.90.0"
-declare -x USER="pcasagra"
-declare -x USERNAME="pcasagra"
-declare -x VSCODE_GIT_ASKPASS_EXTRA_ARGS=""
-declare -x VSCODE_GIT_ASKPASS_MAIN="/usr/share/code/resources/app/extensions/git/dist/askpass-main.js"
-declare -x VSCODE_GIT_ASKPASS_NODE="/usr/share/code/code"
-declare -x VSCODE_GIT_IPC_HANDLE="/run/user/1000/vscode-git-806976d2ff.sock"
-declare -x WAYLAND_DISPLAY="wayland-0"
-declare -x XAUTHORITY="/run/user/1000/.mutter-Xwaylandauth.RHKKT2"
-declare -x XDG_CONFIG_DIRS="/etc/xdg/xdg-ubuntu:/etc/xdg"
-declare -x XDG_CURRENT_DESKTOP="Unity"
-declare -x XDG_DATA_DIRS="/usr/share/ubuntu:/usr/local/share/:/usr/share/:/var/lib/snapd/desktop"
-declare -x XDG_MENU_PREFIX="gnome-"
-declare -x XDG_RUNTIME_DIR="/run/user/1000"
-declare -x XDG_SESSION_CLASS="user"
-declare -x XDG_SESSION_DESKTOP="ubuntu"
-declare -x XDG_SESSION_TYPE="wayland"
-declare -x XMODIFIERS="@im=ibus"
-declare -x a="\"quotes\""
+declare -x SHELL=/bin/bash
+declare -x SESSION_MANAGER=local/pcasagra-VirtualBox:@/tmp/.ICE-unix/1346,unix/pcasagra-VirtualBox:/tmp/.ICE-unix/1346
+declare -x QT_ACCESSIBILITY=1
+declare -x COLORTERM=truecolor
+declare -x XDG_CONFIG_DIRS=/etc/xdg/xdg-ubuntu:/etc/xdg
+declare -x SSH_AGENT_LAUNCHER=gnome-keyring
+declare -x XDG_MENU_PREFIX=gnome-
+declare -x TERM_PROGRAM_VERSION=1.90.0
+declare -x GNOME_DESKTOP_SESSION_ID=this-is-deprecated
+declare -x LC_ADDRESS=de_DE.UTF-8
+declare -x GNOME_SHELL_SESSION_MODE=ubuntu
+declare -x LC_NAME=de_DE.UTF-8
+declare -x SSH_AUTH_SOCK=/run/user/1000/keyring/ssh
+declare -x XMODIFIERS=@im=ibus
+declare -x DESKTOP_SESSION=ubuntu
+declare -x LC_MONETARY=de_DE.UTF-8
+declare -x GTK_MODULES=gail:atk-bridge
+declare -x PWD=/home/pcasagra/42/minishell_brunos_branch/test/temp
+declare -x LOGNAME=pcasagra
+declare -x XDG_SESSION_DESKTOP=ubuntu
+declare -x XDG_SESSION_TYPE=wayland
+declare -x SYSTEMD_EXEC_PID=1383
+declare -x XAUTHORITY=/run/user/1000/.mutter-Xwaylandauth.RHKKT2
+declare -x VSCODE_GIT_ASKPASS_NODE=/usr/share/code/code
+declare -x IM_CONFIG_CHECK_ENV=1
+declare -x GJS_DEBUG_TOPICS=JS ERROR;JS LOG
+declare -x HOME=/home/pcasagra
+declare -x USERNAME=pcasagra
+declare -x IM_CONFIG_PHASE=1
+declare -x LANG=en_US.UTF-8
+declare -x LC_PAPER=de_DE.UTF-8
+declare -x LS_COLORS=rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.zst=01;31:*.tzst=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.wim=01;31:*.swm=01;31:*.dwm=01;31:*.esd=01;31:*.jpg=01;35:*.jpeg=01;35:*.mjpg=01;35:*.mjpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.webp=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36:
+declare -x XDG_CURRENT_DESKTOP=Unity
+declare -x WAYLAND_DISPLAY=wayland-0
+declare -x GIT_ASKPASS=/usr/share/code/resources/app/extensions/git/dist/askpass.sh
+declare -x INVOCATION_ID=e94f03fe0a6c4e15909f802cd253c84f
+declare -x MANAGERPID=1201
+declare -x CHROME_DESKTOP=code-url-handler.desktop
+declare -x GJS_DEBUG_OUTPUT=stderr
+declare -x VSCODE_GIT_ASKPASS_EXTRA_ARGS=
+declare -x GNOME_SETUP_DISPLAY=:1
+declare -x LESSCLOSE=/usr/bin/lesspipe %s %s
+declare -x XDG_SESSION_CLASS=user
+declare -x LC_IDENTIFICATION=de_DE.UTF-8
+declare -x TERM=xterm-256color
+declare -x LESSOPEN=| /usr/bin/lesspipe %s
+declare -x USER=pcasagra
+declare -x VSCODE_GIT_IPC_HANDLE=/run/user/1000/vscode-git-806976d2ff.sock
+declare -x DISPLAY=:0
+declare -x LC_TELEPHONE=de_DE.UTF-8
+declare -x QT_IM_MODULE=ibus
+declare -x LC_MEASUREMENT=de_DE.UTF-8
+declare -x XDG_RUNTIME_DIR=/run/user/1000
+declare -x LC_TIME=de_DE.UTF-8
+declare -x VSCODE_GIT_ASKPASS_MAIN=/usr/share/code/resources/app/extensions/git/dist/askpass-main.js
+declare -x JOURNAL_STREAM=8:7897
+declare -x XDG_DATA_DIRS=/usr/share/ubuntu:/usr/local/share/:/usr/share/:/var/lib/snapd/desktop
+declare -x GDK_BACKEND=x11
+declare -x PATH=/usr/local/nodejs/bin:/home/pcasagra/.local/bin:/usr/local/nodejs/bin:/home/pcasagra/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin:/home/pcasagra/clion-2024.1.2/bin/clion.sh:/home/pcasagra/clion-2024.1.2/bin/clion.sh
+declare -x GDMSESSION=ubuntu
+declare -x ORIGINAL_XDG_CURRENT_DESKTOP=ubuntu:GNOME
+declare -x DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/1000/bus
+declare -x GIO_LAUNCHED_DESKTOP_FILE_PID=2342
+declare -x GIO_LAUNCHED_DESKTOP_FILE=/usr/share/applications/code.desktop
+declare -x LC_NUMERIC=de_DE.UTF-8
+declare -x OLDPWD=/home/pcasagra/42/minishell_brunos_branch/test
+declare -x TERM_PROGRAM=vscode
+declare -x a="quotes"
 
 ======== stderr =======
 
 ======== return =======
 return value: 0

==> diffs/questionable_export_spaces1.diff <==
diff of execution of:
echo 'export a=\'    \'; echo $a' | tr ';' '\n' | [shell]
--- expected	2024-09-01 20:03:58.941668055 +0200
+++ found	2024-09-01 20:03:59.000668112 +0200
@@ -1,6 +1,6 @@
-
+    
 
 ======== stderr =======
 
 ======== return =======
 return value: 0

==> diffs/questionable_export_spaces2.diff <==
diff of execution of:
echo 'export a="a    b"; echo $a' | tr ';' '\n' | [shell]
--- expected	2024-09-01 20:03:59.096668206 +0200
+++ found	2024-09-01 20:03:59.146668255 +0200
@@ -1,6 +1,6 @@
-a b
+a    b
 
 ======== stderr =======
 
 ======== return =======
 return value: 0

==> diffs/questionable_glob_order.diff <==
diff of execution of:
echo 'touch a b c A B C aa ab ac Aa Ab Ac aA aB aC; echo *' | tr ';' '\n' | [shell]
--- expected	2024-09-01 20:03:59.458668561 +0200
+++ found	2024-09-01 20:03:59.507668609 +0200
@@ -1,6 +1,6 @@
-a A aa aA Aa ab aB Ab ac aC Ac b B c C infile readonly
+Ac a aC . aB infile aa .. readonly C A B Aa c ac ab aA b Ab
 
 ======== stderr =======
 
 ======== return =======
 return value: 0

==> diffs/questionable_pwd_rmdir.diff <==
diff of execution of:
echo 'mkdir tmp; cd tmp; rmdir $PWD; pwd' | tr ';' '\n' | [shell]
--- expected	2024-09-01 20:04:00.779669852 +0200
+++ found	2024-09-01 20:04:00.827669899 +0200
@@ -1,6 +1,5 @@
-/home/pcasagra/42/minishell_brunos_branch/test/temp/tmp
 
 ======== stderr =======
-
+getcwd() error
 ======== return =======
-return value: 0
+return value: 1

==> diffs/questionable_redir_ambiguous.diff <==
diff of execution of:
echo 'cat < *;' | tr ';' '\n' | [shell]
--- expected	2024-09-01 20:03:59.664668762 +0200
+++ found	2024-09-01 20:03:59.728668825 +0200
@@ -1,5 +1,7 @@
+infile content
+infile second line
 
 ======== stderr =======
-[progname]: *: ambiguous redirect
+[progname]: ..: Is a directory
 ======== return =======
-return value: 1
+return value: 0

==> diffs/questionable_SHLVL.diff <==
diff of execution of:
echo 'echo $SHLVL' | tr ';' '\n' | [shell]
--- expected	2024-09-01 20:03:59.253668360 +0200
+++ found	2024-09-01 20:03:59.316668422 +0200
@@ -1,6 +1,6 @@
-2
+1
 
 ======== stderr =======
 
 ======== return =======
 return value: 0

==> diffs/redir_expand_cmd.diff <==
diff of execution of:
echo '$fail < failed' | tr ';' '\n' | [shell]
--- expected	2024-09-01 20:04:20.319688559 +0200
+++ found	2024-09-01 20:04:20.546688772 +0200
@@ -1,5 +1,5 @@
 
 ======== stderr =======
-[progname]: failed: No such file or directory
+
 ======== return =======
 return value: 1

==> diffs/redir_heredoc2.diff <==
diff of execution of:
echo 'grep o << EOF > out;hello;there;EOF; ls -l --time-style=+' | tr ';' '\n' | [shell]
--- expected	2024-09-01 20:04:16.616685071 +0200
+++ found	2024-09-01 20:04:16.734685182 +0200
@@ -1,9 +1,11 @@
 total 8
 -rw-rw-r-- 1 pcasagra pcasagra 34  infile
 -rw-rw-r-- 1 pcasagra pcasagra  6  out
 -r--r--r-- 1 pcasagra pcasagra  0  readonly
 
 ======== stderr =======
-
+[progname]: hello: command not found
+[progname]: there: command not found
+[progname]: EOF: command not found
 ======== return =======
 return value: 0

==> diffs/redir_heredoc.diff <==
diff of execution of:
echo 'cat << EOF;hello;there;EOF' | tr ';' '\n' | [shell]
--- expected	2024-09-01 20:04:16.456684919 +0200
+++ found	2024-09-01 20:04:16.517684977 +0200
@@ -1,7 +1,9 @@
 hello
 there
 
 ======== stderr =======
-
+[progname]: hello: command not found
+[progname]: there: command not found
+[progname]: EOF: command not found
 ======== return =======
 return value: 0

==> diffs/redir_multi_here.diff <==
diff of execution of:
echo 'cat <<EOF <<OH;i;am;here;EOF;oh;OH' | tr ';' '\n' | [shell]
--- expected	2024-09-01 20:04:23.162691219 +0200
+++ found	2024-09-01 20:04:23.171691227 +0200
@@ -1,6 +1,11 @@
 oh
 
 ======== stderr =======
-
+[progname]: i: command not found
+[progname]: am: command not found
+[progname]: here: command not found
+[progname]: EOF: command not found
+[progname]: oh: command not found
+[progname]: OH: command not found
 ======== return =======
 return value: 0

==> diffs/return_exit_ko1.diff <==
diff of execution of:
echo 'exit one two; echo $?' | tr ';' '\n' | [shell]
--- expected	2024-09-01 20:04:25.957693824 +0200
+++ found	2024-09-01 20:04:26.063693922 +0200
@@ -1,5 +1,5 @@
 
 ======== stderr =======
-[progname]: exit: one: numeric argument required
+[progname]: exit: too many arguments
 ======== return =======
 return value: 1

==> diffs/return_exit_ko2.diff <==
diff of execution of:
echo 'exit 1 2; echo $?' | tr ';' '\n' | [shell]
--- expected	2024-09-01 20:04:26.125693979 +0200
+++ found	2024-09-01 20:04:26.165694016 +0200
@@ -1,6 +1,5 @@
-1
 
 ======== stderr =======
 [progname]: exit: too many arguments
 ======== return =======
-return value: 0
+return value: 1

==> diffs/return_ok2.diff <==
diff of execution of:
echo 'echo hi; echo thisiaavariable$?thatneedsexpansion' | tr ';' '\n' | [shell]
--- expected	2024-09-01 20:04:25.295693208 +0200
+++ found	2024-09-01 20:04:25.379693286 +0200
@@ -1,7 +1,7 @@
 hi
-thisiaavariable0thatneedsexpansion
+thisiaavariable 0 thatneedsexpansion
 
 ======== stderr =======
 
 ======== return =======
 return value: 0

==> diffs/special_reset_status.diff <==
diff of execution of:
echo 'cat fail; $FAIL; echo $?' | tr ';' '\n' | [shell]
--- expected	2024-09-01 20:04:00.969670038 +0200
+++ found	2024-09-01 20:04:01.026670093 +0200
@@ -1,6 +1,6 @@
-0
+1
 
 ======== stderr =======
 [progname]: fail: No such file or directory
 ======== return =======
 return value: 0
